### DISCUSSION TITLE
useQuery and "Cannot update a component while rendering a different component" error

### DISCUSSION THREAD

**Discussion by yngwi** [urql-graphql/urql#2278] [Q&A]
Hi,

I'm currently using urql in my app together with query hooks auto generated by *@graphql-codegen*. Generally speaking it works fine but I'm getting now a error message, `Warning: Cannot update a component (EntityList) while rendering a different component (TagList). To locate the bad setState() call inside TagList, follow the stack trace as described in https://reactjs.org/link/setstate-in-render`, by react whose stack points to the urql `useQuery` hook used in the autogenerated code:

```typescript
export function useItemTagsQuery(options: Omit<Urql.UseQueryArgs<ItemTagsQueryVariables>, 'query'>) {
  return Urql.useQuery<ItemTagsQuery>({ query: ItemTagsDocument, ...options });
};
```
As far as I can tell the usage is correct and the component that react complains about doesn't use other state-handling related hooks apart from the auto-generated `useItemTagsQuery` quoted above. Can you help me figure out how to get rid of this error message.

Thanks!
Daniel

**Comment by eschaefer** [comment]
I managed to find a workaround for this using this https://github.com/FormidableLabs/urql/issues/1382#issuecomment-1030701320

**Reply by kitten** [reply]
Ehem 😅 you may just want to read this potentially: https://github.com/FormidableLabs/urql/issues/2309

**Reply by eschaefer** [reply]
Oh wow ok thank you for the heads-up!

**Reply by yngwi** [reply]
Thanks for the heads up!

**Reply by Kishore180994** [reply]
any updates on this? Is this issue resolved?

**Reply by kitten** [reply]
@Kishore180994 Yes, this issue should now be fully resolved in all cases with this patch: https://github.com/FormidableLabs/urql/pull/2556 in `urql@2.2.3` and upwards

**Comment by ajtosh77** [comment]
I am still seeing the "Cannot update a component" warning from useQuery (i'm on version 3.0.3); same sort of usage as you @yngwi. Did the warning get resolved for you, @yngwi?

**Reply by yngwi** [reply]
Sorry, I couldn't try it yet, I've been working on a different a project for some time where we don't use graphql. I'll try to find out.

**Comment by strblr** [comment]
I have the same warnings popping up all over the place in my app.
